javaparser-symbol-solver-logic/src/main/java/com/github/javaparser/symbolsolver/logic/TypeDeclarationWithResolutionCapabalities.java

```diff
@@ -0,0 +1,16 @@
+package com.github.javaparser.symbolsolver.logic;
+
+import com.github.javaparser.resolution.declarations.ResolvedMethodDeclaration;
+import com.github.javaparser.resolution.types.ResolvedType;
+import com.github.javaparser.symbolsolver.model.resolution.SymbolReference;
+import com.github.javaparser.symbolsolver.model.resolution.TypeSolver;
+
+import java.util.List;
+
+public interface TypeDeclarationWithResolutionCapabalities {
+
+	SymbolReference<ResolvedMethodDeclaration> solveMethod(String name, List<ResolvedType> argumentsTypes,
```

Comment:
Can we avoid using the typesolver as a parameter?

Reply:
Yea, I've been looking into this and it requires surprisingly many changes, as the removal of that parameter propagates quite far. I am currently working on making it work - would definitely make some of these methods nicer, especially with a field available most of the time, as you have noted on some other files.